### PR TITLE
AUT-607: Create SMS validator for updating phone number

### PR DESCRIPTION
**Pull request by ayoshebby**
## What?

Create new SMS validator and refactor UpdatePhoneNumber handler to make use of the sms validator. This is required to validate OTP code when invalid or code entered too many times. These codes will then determine the routes for error messages in the accounts management page

## Why?

Required so that the error codes returned by the handler is reflected across pages for account management

## Related PRs

[Build UI components](https://github.com/alphagov/di-authentication-account-management/pull/561)

### CODE REVIEW

File: account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/UpdatePhoneNumberHandler.java [hunk 1]
```diff
@@ -12,20 +12,17 @@
...
 import uk.gov.di.authentication.shared.helpers.RequestBodyHelper;
 import uk.gov.di.authentication.shared.helpers.RequestHeaderHelper;
 import uk.gov.di.authentication.shared.serialization.Json;
 import uk.gov.di.authentication.shared.serialization.Json.JsonException;
-import uk.gov.di.authentication.shared.services.AuditService;
-import uk.gov.di.authentication.shared.services.ConfigurationService;
-import uk.gov.di.authentication.shared.services.DynamoService;
-import uk.gov.di.authentication.shared.services.RedisConnectionService;
-import uk.gov.di.authentication.shared.services.SerializationService;
+import uk.gov.di.authentication.shared.services.*;
```
Comment (JHjava) [inline account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/UpdatePhoneNumberHandler.java:24]: Try and avoid the wildcard imports

File: account-management-api/src/test/java/uk/gov/di/accountmanagement/lambda/UpdatePhoneNumberHandlerTest.java [hunk 2]
```diff
@@ -135,15 +152,52 @@ public void shouldReturnErrorWhenOtpCodeIsNotValid() throws Json.JsonException {
...
+        when(codeStorageService.isBlockedForEmail(any(), any())).thenReturn(true);
+
+        APIGatewayProxyResponseEvent result = handler.handleRequest(event, context);
+
+        assertThat(result, hasStatus(400));
+        verify(dynamoService, times(0)).updatePhoneNumber(EMAIL_ADDRESS, INVALID_PHONE_NUMBER);
+        NotifyRequest notifyRequest = new NotifyRequest(INVALID_PHONE_NUMBER, PHONE_NUMBER_UPDATED);
+        verify(sqsClient, times(0)).send(objectMapper.writeValueAsString(notifyRequest));
+        String expectedResponse = objectMapper.writeValueAsString(ErrorResponse.ERROR_1027);
         assertThat(result, hasBody(expectedResponse));
```
Comment (JHjava) [inline account-management-api/src/test/java/uk/gov/di/accountmanagement/lambda/UpdatePhoneNumberHandlerTest.java:201]: could do something like `assertThat(result, hasJsonBody(ErrorResponse.ERROR_1027));` instead

File: shared/src/main/java/uk/gov/di/authentication/shared/validation/SMSCodeValidator.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+        if (hasExceededRetryLimit()) {
+            LOG.info("Exceeded code retry limit");
+            return Optional.of(ErrorResponse.ERROR_1027);
+        }
+
+        boolean isValidOtp = isCodeValid(code, NotificationType.VERIFY_PHONE_NUMBER);
+
+        if (!isValidOtp) {
+            LOG.info("In");
```
Comment (JHjava) [inline shared/src/main/java/uk/gov/di/authentication/shared/validation/SMSCodeValidator.java:41]: Need something adding to this log message

**Review by JHjava — COMMENTED** [review]
Looks good. Just a few minir comments

**Comment by dbes-gds** [comment]
Might be good to keep an eye on the sonarcloud issues as well, and fix them if you can even if they look a bit trivial

**Review by JHjava — APPROVED** [review]

**Pull request merged**